### ISSUE TITLE
AWSDate scalar serialize may fail if incoming string date in full format

### ISSUE THREAD

**Issue by katesclau** [aws-amplify/amplify-cli#8139]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

As dependencie to https://github.com/bboure/serverless-appsync-simulator

### If applicable, what version of Node.js are you using?

v14.17.0

### Amplify CLI Version

"amplify-appsync-simulator": "^1.27.4",

### What operating system are you using?

Ubuntu

### Amplify Categories

Not applicable

### Amplify Commands

Not applicable

### Describe the bug

Context:
We use https://github.com/bboure/serverless-appsync-simulator  to run serverless in offline mode and develop AppSync features (VTLs, DataSources, Lambdas, etc)

While working with our current set of resolvers, we've noticed that AWSDate parsing in `amplify-appsync-simulator` can fail  when AWSDate coming from RDS DataSource, due to full string format provided - ie.: `2021-09-09 00:00:00.000Z`.

Solution we took was a fork on `amplify-appsync-simulator` [here](https://github.com/returnmates/amplify-appsync-simulator) and adapting as [such](https://github.com/returnmates/amplify-appsync-simulator/commit/a7e72d2105ed0518ca51191634817a83cad19cf1#r56206524)

### Expected behavior

Parses AWSDate string coming from RDS Datasources

### Reproduction steps

1. Create a serverless project with serverless-appsync-plugin & serverless-appsync-simulator
2. Add an RDS DataSource
3. Create a Query that returns AWSDate from RDS DataSource
4. Run serverless offline
5. Query

### GraphQL schema(s)

<details>

```graphql
schema {
  query: Query
}

type Query {
  today: AWSDate!
}
```
</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by cjihrig** [comment]
It looks like there are a few different things going on here:

First, the current AppSync simulator approach uses `GraphQLDate` from the `graphql-iso-date` package. This does not work with all formats supported by AppSync. Based on [these docs](https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html), we need to support a few different value types that `GraphQLDate` does not support:

- `1970-01-01` (this is currently supported)
- `1970-01-01Z`
- `1970-01-01-07:00`
- `1970-01-01+05:30`

The other problem is compatibility with RDS dates. I've seen some issues like [this one from Stack Overflow](https://stackoverflow.com/questions/58144742/unable-to-serialise-awsdate-from-rds-in-appsync-response-mapping-template) where the format returned by RDS was not compatible with AppSync. That issue is a few years old at this point, so maybe things have changed/improved. Does your problem reproduce in AppSync, or do you only see it in the simulator?

**Comment by katesclau** [comment]
> Does your problem reproduce in AppSync, or do you only see it in the simulator?
Yes, but this is mostly an issue with the simulator itself, not on AppSync.
